NEWS.md

```diff
@@ -1,186 +1,296 @@
+   + Added more explanation why certain functions might be undesirable and what alternatives to use;
+     ditto for `undesirable_operator_linter()` (#1133, #1146, #1159, @AshesITR)
+* `assignment_linter()`: extended and add arguments (#915, @michaelchirico)
+   + right assignments are now linted by default (`->` and `->>`)
+   + new argument `allow_cascading_assign` (`TRUE` by default) toggles whether to lint `<<-` and `->>`
+   + new argument `allow_right_assign` (`FALSE` by default) toggles whether to lint `->` and `->>`
+* `infix_spaces_linter()`
+   + added argument `allow_multiple_spaces` (`TRUE` by default) which toggles
+     whether to generate a lint for operators used with multiple spaces, e.g. `x   +   2`.
+     The default setting allows extra spacing to be used to increase
+     line-to-line alignment (#940, @f-ritter and @michaelchirico)
+   + extended so that usages like `a~b` and `function(a=1) { ... }` are linted (#930, #michaelchirico)
+   + added argument `exclude_operators` to disable lints on selected infix operators.
+     By default, all "low-precedence" operators throw lints; see `?infix_spaces_linter` for an enumeration of these.
+     (#914, @michaelchirico)
+   + add exception for `box::use()` declarations (#1087, @klmr)
+* `trailing_whitespace_linter()`
+   + extended to also lint completely blank lines by default (#1044, @AshesITR)
+   + added argument `allow_empty_lines` (`FALSE` by default) to toggle this behavior
+   + improved so that trailing whitespace inside string literals does not trigger a lint (#1045, @AshesITR)
+   * added argument `allow_in_strings` (`TRUE` by default) to toggle this behavior
+* `trailing_blank_lines_linter()`: extend to lint files without a terminal newline (#675, @AshesITR)
+* `object_name_linter()`
+   + improved generic detection -- in user-defined method `my_method.upstream.class`,
+     `upstream.class` no longer throws a lint because the generic (`my_method`)
+     properly uses `snake_case` (#737, @AshesITR)
+   + extended to exclude special R namespace hook functions such as `.onLoad` (#500, #614, @AshesITR and @michaelchirico)
+   + extended to correctly detect imported functions when linting packages (#642, @AshesITR)
+   + correctly detect assignment generics like `names<-.class_name` (#843, @jonkeane)
+   + added new styles `"symbols"` and `"SNAKE_CASE"` (#494, #495, #615, #670, @michaelchirico and @AshesITR)
+* `object_usage_linter()`
+   + detect global variables if there are top-level dollar-assignments (#666, #709, @AshesITR)
+   + report usage warnings spanning multiple lines (#507, @AshesITR)
+   + detect usages inside `glue::glue()` constructs (#942, @AshesITR)
+   + detect within functions assigned with `=` instead of `<-` (#1081, @michaelchirico)
+   + detect functions exported by packages that are explicitly attached using `library()` or
+     `require()` calls (#1127, @AshesITR)
+   + improved location information in some cases (#1285, @AshesITR)
```

Comment:
@AshesITR could you expand on this? someone reading the NEWS reads this and thinks "well, what cases?"

it's also not clear from the #1285 description -- someone reading this bullet would have to read the full PR diff to get an idea of what's changed exactly. better to give a specific example.